### PR TITLE
Implements remaining HAPPYCHAT_IO_RECEIVE_* actions

### PR DESCRIPTION
**Pull request by nosolosw**
Master issue: https://github.com/Automattic/wp-calypso/issues/18670

This PR builds upon #19251 and implements all the remaining HAPPYCHAT_IO_RECEIVE inbound actions. 19251 should be merged first, and it's set as the base branch for ease of review.

### Test

`npm run test-client happychat`

Manually start a happychat session and test that it works normally (it can connect, messages are received, audio for receiving messages work, etc).

 ---

_PS: thanks for reviewing effort and helping me get out of this dungeon; there were dragons, but we got out of here[*](https://www.youtube.com/watch?v=39LRMQozNd4). Ahead to the next level!_  :tada:

### CODE REVIEW

File: client/components/happychat/notices.jsx [hunk 1]
```diff
@@ -36,16 +41,19 @@ export class Notices extends Component {
...
-			// Fall through to the same notice as `disconnected`
-			case 'disconnected':
+			case HAPPYCHAT_CONNECTION_STATUS_RECONNECTING:
+			case HAPPYCHAT_CONNECTION_STATUS_DISCONNECTED:
 				return translate(
 					"We're having trouble connecting to chat. Please bear with us while we try to reconnect…"
 				);
+			case HAPPYCHAT_CONNECTION_STATUS_UNAUTHORIZED:
+				return translate(
+					'Your user is not authorized to chat at the moment. Would you mind contacting support by other means?'
```
Comment (southp) [inline client/components/happychat/notices.jsx]: I personally felt a bit confused when I read "Your user" here. Maybe "Your account" or simply "You are not authorized ..." would be easier to understand? I'm not quite confident here so I'm labeling this for copy review :)

**Comment by southp** [comment]
I've tested a few scenarios:

* Send messages back and forth.
* A user goes offline and reconnect.
* An operator goes offline and reconnect.
* The HC server goes offline and reconnect.

and they all works for me. The code also LGTM, but I have labeled it for further copy review since I have a little doubt about the newly-added error message.

Once that is resolved, I'm happy to see it in production 🚢

File: client/components/happychat/notices.jsx (hunk 1 above)
Comment (nosolosw) [reply client/components/happychat/notices.jsx]: Improved messaging in e7c2691

Comment (nosolosw) [reply client/components/happychat/notices.jsx]: Updated text in adf6707 as discussed with Krista in p1509538277000157-slack-editorial

Go to merge this in.

**Pull request merged**